src/NServiceBus.Core.Tests/Msmq/MsmqQueueCreatorTests.cs

```diff
@@ -247,12 +229,16 @@ void DeleteQueueIfPresent(string queueName)
         {
             var path = MsmqAddress.Parse(queueName).PathWithoutPrefix;
 
-            if (!MessageQueue.Exists(path))
+            if (MessageQueue.Exists(path))
             {
-                return;
+                MessageQueue.Delete(path);
             }
+        }
 
-            MessageQueue.Delete(path);
+        string GetReallyLongQueueName()
+        {
+            var maxQueueNameForSetPermissionToWork = 102 - Environment.MachineName.Length;
```

Comment:
@ramonsmits - Can you provide the significance of 102. Can you add some docs link? Also, I simplified this method because, I didn't think we needed to generate a new Guid every time, given that this is getting padded with a's and we always start off with a non-existent queue.  Ideally, rather than this method, it just should be a var with a really long name with the number of characters, IMO. cc / @Particular/msmq-transport-maintainers @Particular/nservicebus-maintainers